addons/account/models/res_partner_bank.py

```diff
@@ -1,16 +1,234 @@
+                        'structured_communication': structured_communication,
+                    }
+
+                elif not silent_errors:
+                    error_header = _("The following error prevented '%s' QR-code to be generated though it was detected as eligible: ", candidate_name)
+                    raise UserError(error_header + error_message)
+
+        return None
+
+    def build_qr_code_url(self, amount, free_communication, structured_communication, currency, debtor_partner, qr_method=None, silent_errors=True):
+        vals = self._build_qr_code_vals(amount, free_communication, structured_communication, currency, debtor_partner, qr_method, silent_errors)
+        if vals:
+            return self._get_qr_code_url(
+                vals['qr_method'],
+                vals['amount'],
+                vals['currency'],
+                vals['debtor_partner'],
+                vals['free_communication'],
+                vals['structured_communication'],
+            )
+        return None
+
+    def build_qr_code_base64(self, amount, free_communication, structured_communication, currency, debtor_partner, qr_method=None, silent_errors=True):
+        vals = self._build_qr_code_vals(amount, free_communication, structured_communication, currency, debtor_partner, qr_method, silent_errors)
+        if vals:
+            return self._get_qr_code_base64(
+                vals['qr_method'],
+                vals['amount'],
+                vals['currency'],
+                vals['debtor_partner'],
+                vals['free_communication'],
+                vals['structured_communication']
+            )
+        return None
+
+    def _get_qr_vals(self, qr_method, amount, currency, debtor_partner, free_communication, structured_communication):
+        return None
+
+    def _get_qr_code_generation_params(self, qr_method, amount, currency, debtor_partner, free_communication, structured_communication):
+        return None
+
+    def _get_qr_code_url(self, qr_method, amount, currency, debtor_partner, free_communication, structured_communication):
+        """ Hook for extension, to support the different QR generation methods.
+        This function uses the provided qr_method to try generation a QR-code for
+        the given data. It it succeeds, it returns the report URL to make this
+        QR-code; else None.
+
+        :param qr_method: The QR generation method to be used to make the QR-code.
+        :param amount: The amount to be paid
+        :param currency: The currency in which amount is expressed
+        :param debtor_partner: The partner to which this QR-code is aimed (so the one who will have to pay)
+        :param free_communication: Free communication to add to the payment when generating one with the QR-code
+        :param structured_communication: Structured communication to add to the payment when generating one with the QR-code
+        """
+        # pylint: disable=E1137
```

Comment:
use the lint names, not the meaningless and impossible to understand error codes.